api/common_test.go

```diff
@@ -0,0 +1,175 @@
+package api
+
+import (
+	"io/ioutil"
+	"path/filepath"
+	"testing"
+
+	"github.com/docker/docker/api/types"
+)
+
+type ports []types.Port
+
+// DisplayablePorts
+func TestDisplayablePorts(t *testing.T) {
+	cases := map[*ports]string{
+		&ports{{
+			PrivatePort: 9988,
+			Type:        "tcp",
+		}}: "9988/tcp",
+		&ports{{
+			PrivatePort: 9988,
+			Type:        "udp",
+		}}: "9988/udp",
+		&ports{{
+			IP:          "0.0.0.0",
+			PrivatePort: 9988,
+			Type:        "tcp",
+		}}: "0.0.0.0:0->9988/tcp",
+		&ports{{
+			PrivatePort: 9988,
+			PublicPort:  8899,
+			Type:        "tcp",
+		}}: "9988/tcp",
+		&ports{{
+			IP:          "4.3.2.1",
+			PrivatePort: 9988,
+			PublicPort:  8899,
+			Type:        "tcp",
+		}}: "4.3.2.1:8899->9988/tcp",
+		&ports{{
+			IP:          "4.3.2.1",
+			PrivatePort: 9988,
+			PublicPort:  9988,
+			Type:        "tcp",
+		}}: "4.3.2.1:9988->9988/tcp",
+		&ports{{
+			PrivatePort: 9988,
+			Type:        "udp",
+		}, {
+			PrivatePort: 9988,
+			Type:        "udp",
+		}}: "9988/udp, 9988/udp",
+		&ports{{
+			IP:          "1.2.3.4",
+			PublicPort:  9998,
+			PrivatePort: 9998,
+			Type:        "udp",
+		}, {
+			IP:          "1.2.3.4",
+			PublicPort:  9999,
+			PrivatePort: 9999,
+			Type:        "udp",
+		}}: "1.2.3.4:9998-9999->9998-9999/udp",
+		&ports{{
+			IP:          "1.2.3.4",
+			PublicPort:  8887,
+			PrivatePort: 9998,
+			Type:        "udp",
+		}, {
+			IP:          "1.2.3.4",
+			PublicPort:  8888,
+			PrivatePort: 9999,
+			Type:        "udp",
+		}}: "1.2.3.4:8887->9998/udp, 1.2.3.4:8888->9999/udp",
```

Comment:
This one was *fun*, I though it would do ``1.2.3.4:8887-8888->9998-9999/udp`` at first, but did not :sweat_smile: .